packages/grafana-data/src/transformations/transformers/joinDataFrames.ts

```diff
@@ -362,6 +362,74 @@ function joinOuterTabular(
   return data;
 }
 
+/**
+ * This function performs a sql-style inner join on tabular data;
+ * it will combine records from two tables whenever there are matching
+ * values in a field common to both tables.
+ *
+ * NOTE: This function implicitly assumes that the first array in each AlignedData
+ * contains the values to join on. It doesn't explicitly specify a column field to join on,
+ * but rather uses the 0th position of the arrays (AlignedData[0]) to determine the joining keys.
+ * Then, when processing the tables, the function iterates over the values in the `xValues`
+ * (the joining keys) array and checks if the current row `currentRow` already includes the value.
+ * If a matching value is found, it joins the corresponding values from the remaining arrays `yValues`
+ * (all other non-joining key arrays) to create a new row in the joined table.
+ *
+ * @param {AlignedData[]} tables - The tables to join.
+ *
+ * @returns {Array<Array<string | number | null | undefined>>} The joined tables as an array of arrays, where each array represents a row in the joined table.
+ */
+function joinInner(tables: AlignedData[]): Array<Array<string | number | null | undefined>> {
+  const joinedTables: Array<Array<string | number | null | undefined>> = [];
+
+  // Recursive function to perform the inner join.
+  const joinTables = (
+    currentTables: AlignedData[],
+    currentIndex: number,
+    currentRow: Array<string | number | null | undefined>
+  ) => {
+    if (currentIndex === currentTables.length) {
+      // Base case: all tables have been joined, add the current row to the final result.
```

Comment:
data-structures and algorithms!! next challenge will be centering a div

Reply:
Using only floats!!!